encodings/sparse/src/lib.rs

```diff
@@ -113,6 +113,15 @@ impl SparseArray {
     ///
     /// Optionally provided fill value will be respected if the array is less than 90% null.
     pub fn encode(array: &dyn Array, fill_value: Option<Scalar>) -> VortexResult<ArrayRef> {
+        if let Some(fill_value) = fill_value.as_ref() {
```

Comment:
@robert3005 we talked about this. I think it'll fail anyway, but its better to make the error explicit.